Add missing troubleshooting items (#151)
Yaskawa-Global/motoros2: doc/troubleshooting.md

After correcting the configuration, the [changes will need to be propagated to the Yaskawa controller](../README.md#updating-the-configuration).

### Alarm: 8011[23 - 54]

*Example:*

```text
ALARM 8011

>  Failed creating/adding/init *something*

Reviewer: Hmm ..

Could we maybe come up with something a bit nicer? :)

I agree we shouldn't duplicate this entry for all possible subcodes, but perhaps we can come up with a slightly nicer example text?

Author: ```
ALARM 8011
 MotoROS2: Fatal Error
 Failed adding
 Failed to init
 Failed creating
```

?

[23] - [54]
```

*Solution:*

> Ensure that the ROS2 version (Galactic, Humble, etc) is the same for MotoROS2, the micro-ROS Agent and the client PC.
> Additionally, ensure that the ROS2 version is one that is [officially supported](https://github.com/Yaskawa-Global/motoros2#general-requirements) by MotoROS2.

Reviewer: ```suggestion
These alarms are often caused by version incompatibilities between ROS 2 (on the client PC), micro-ROS (as part of MotoROS2) and/or the micro-ROS Agent.

Ensure only compatible versions are used.
As an example: the *Humble* version of MotoROS2 should only be used with ROS 2 *Humble* on the client PC and with the *Humble* version of the micro-ROS Agent.

Please also verify the client PC uses a version of ROS 2 that is [supported by MotoROS2](https://github.com/Yaskawa-Global/motoros2#general-requirements).
```

> If the behavior persists, save a copy of the [debug-listener script](#debug-log-client) output and the `PANELBOX.LOG` from the robot's teach pendant.

Reviewer: Would we want to update all references to the debug logger? This is the first time we link to that section like this.

Open a new issue on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues).
Describe the problem and include the following items:

- `PANELBOX.LOG`
- output from debug listener script (complete and unedited)

> - version of MotoROS2

Reviewer: This should already be clear from `PANELBOX.LOG`.

Author: Yeah, but it's helpful to see if there are any discrepancies between what someone thinks they're using and what's actually installed.

- version of micro-ROS Agent

> - version of ROS2 on client PC

Reviewer: ```suggestion
- version of ROS 2 on client PC
```

> - copy of the MotoROS2 yaml configuration file

Reviewer: ```suggestion
- copy of `motoros2_config.yaml` copied from the robot controller
```

we have some text in the readme detailing how to do this, but it's not in its own section, so we can't directly link to it.

After correcting the configuration, the [changes will need to be propagated to the Yaskawa controller](../README.md#updating-the-configuration).

> ### Alarm: 8011[56 - 58]

Reviewer: This is basically the same as for subcodes `[23 - 54]`, right?

Would it be acceptable to merge this section with the previous one?

Author: Unfortunately, that would make [55] non-sequential. 
That would definitely cause issues for someone like me.

Additionally, ensure that the ROS2 version is one that is [officially supported](https://github.com/Yaskawa-Global/motoros2#general-requirements) by MotoROS2.

If the behavior persists, save a copy of the [debug-listener script](#debug-log-client) output and the `PANELBOX.LOG` from the robot's teach pendant.
Open a new issue on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues).
Describe the problem and include the following items:

- `PANELBOX.LOG`
- output from debug listener script (complete and unedited)
- version of MotoROS2
- version of micro-ROS Agent
- version of ROS2 on client PC
- copy of the MotoROS2 yaml configuration file
- verbatim copy of the alarm text as seen on the teach pendant (alarm number and `[subcode]`).

### Alarm: 8011[60 - 62]

*Example:*

```text
ALARM 8011
 Must enable ETHERNET function
[60 - 61]
```

*Solution:*
The ETHERNET function must be enabled for one (or both) LAN interface in the robot controller.
Please contact your local Yaskawa representative to request this function.

### Alarm: 8011[59]

*Example:*

```text
ALARM 8011
 Host on NIC check x
[59]
```

*Solution:*

> 

Reviewer: should this be moved to before the previous section?

Also: is it complete? :)

Author: I'm not sure what happened here...

After correcting the configuration, the [changes will need to be propagated to the Yaskawa controller](../README.md#updating-the-configuration).

### Alarm: 8011[56 - 58]

Please refer to [Alarm: 8011[23 - 54]](#alarm-801123---54).

### Alarm: 8011[59]

*Example:*

```text
ALARM 8011
 Host on NIC check x
[59]
```

*Solution:*

The `agent_ip_address` key in the `motoros2_config.yaml` configuration file is an address that is not reachable by the robot controller.

Options:

1. Modify the `agent_ip_address` key and specify an IP address that is on the robot's subnet.
   Now follow the instructions [to propagate the changes to the Yaskawa controller](../README.md#updating-the-configuration).
2. Modify the robot controller's IP so it is on the Agent's subnet.
3. Modify the robot controller's network settings to add a gateway which can reach the Agent's IP address.

Refer to the relevant Yaskawa Motoman documentation for information on how to change the controller's network configuration.

### Alarm: 8011[60 - 62]

*Example:*

```text
ALARM 8011
 Must enable ETHERNET function

> [60 - 61]

Reviewer: is it `[60 - 62]` as in the section header, or `[60 - 61]` as stated here.

Author: 62
